### ISSUE TITLE
[Netty 5] HttpClientCodec (and other Codec classes) cleanup / removal

### ISSUE THREAD

**Issue by normanmaurer** [netty/netty#8526]
The `*Codec` classes often provide a lightweight wrapper around the `Encoder`/`Decoder` for the current protocol.
We will remove the `*Codec` classes in cases where the main purpose is just to insert `encoder`/`decoder`.

**Comment by Bennett-Lynch** [comment]
Why?

**Comment by normanmaurer** [comment]
@Bennett-Lynch just because its less code to maintain and it not adds a lot of value.  That said I am all ears to other opinions here.

**Comment by vietj** [comment]
the `HttpServerCodec` maintains state with the `queue` field.

that being said in Vert.x we don't use it because we allow the response to specify whether to specify whether the content is always empty or not.

**Comment by normanmaurer** [comment]
@vietj yes thats why I said `We will remove the *Codec classes in cases where the main purpose is just to insert encoder/decoder.`. We will need to review in case by case basis if there is a good reason to have it or not.

**Comment by vietj** [comment]
@normanmaurer you got me, I'll pay more attention next time :-)

**Comment by Bennett-Lynch** [comment]
I might need to see the actual proposed change list, but I'd just like to call out that the convenience afforded to users should at least be considered before throwing out `*Codec`s.  Convenience in terms of readability, simplicity, etc.  As a Netty user, I want to be able to just install a `HttpClientCodec`, rather than both `HttpResponseDecoder` and `HttpRequestEncoder`.  If the maintainability by Netty contributors is the concern, maybe that's a small worth paying.

**Comment by normanmaurer** [comment]
@Bennett-Lynch fair enough and we will take this for sure into account.

**Comment by chrisvest** [comment]
@vietj @Bennett-Lynch @normanmaurer Sounds like the codec is very convenient and helpful for the common case, while some more advanced uses need to use separate encoder and decoder.

So… is there anything we need to do here, or do we just close this?

**Comment by chrisvest** [comment]
Let's close this.
